### DISCUSSION TITLE
Is it possible to update the cellular modem firmware with the 2 actions in follow description?

### DISCUSSION THREAD

**Discussion by cihun** [fwupd/fwupd#6059] [Q&A]
1. In LVFS, we plan to upload the modem firmware package which include programs for managing modems and firmware images of modem. 

2. In fwupd, we plan to develop a plug-in for fibocom modem using to copy various files from the cab package to a specified directory and run programs from cab package.

Please check the feasibility of these two points.

Thanks.

**Comment by hughsie** [comment]
You can't include the flasher as part of the firmware payload. All the executable code needs to live in fwupd.

**Reply by cihun** [reply]
Hi

We see the following on this page (<https://lvfs.readthedocs.io/en/latest/intro.html#fwupd>): 

> Other files are permitted in the archive which allows the same deliverable to be used for the Windows Update system.

Can we store some binary programs in the `.cab` package?

Thanks.

**Reply by hughsie** [reply]
> Can we store some binary programs in the .cab package?

No, see https://lvfs.readthedocs.io/en/latest/custom-plugin.html

**Reply by superm1** [reply]
Distribution in the cab and using it are two different things. So you want to have same cab file for both OS, but only use the payload and xml on Linux but use the executable on Windows?

I wonder if we can support this but just filter out anything not used when it's uploaded to LVFS @hughsie

**Reply by hughsie** [reply]
I'd need to go back to the lawyers if we start including win32.exe stuff.

**Reply by superm1** [reply]
That's why I'm suggesting filtering it at LVFS level when uploaded. LVFS signs and rebuilds cab anyway.

**Reply by hughsie** [reply]
Right, I guess you're correct -- we'd do that now I think -- i.e. just leave the .xml, .inf, payload and strip out anything else.
